### PR TITLE
[Merged by Bors] - feat port: data.set.sigma

### PR DESCRIPTION
**Pull request by PatrickMassot**
fc2ed6f838ce7c9b7c7171e58d78eaf7b438fb0e

### CODE REVIEW

**Comment by PatrickMassot** [comment]
Here I have a question about the naming convention, asked at https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/referring.20to.20namespace.20in.20theorem.20name. The last commit in this PR implements the different names, it can be easily reverted.

File: Mathlib/Data/Set/Sigma.lean [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Data.Set.Image
+
+/-!
+# Sets in sigma types
+
+This file defines `set.sigma`, the indexed sum of sets.
```
Comment (semorrison) [inline Mathlib/Data/Set/Sigma.lean:11]:
```suggestion
This file defines `Set.sigma`, the indexed sum of sets.
```

**Comment by PatrickMassot** [comment]
bors merge

**Comment by PatrickMassot** [comment]
bors merge

**Comment by PatrickMassot** [comment]
bors merge

**Pull request closed without merging**